test/EventManagerFactoryTest.php

```diff
@@ -130,80 +160,143 @@ public function testInstanceListener() : void
     }
 
     /**
      * @param mixed $subscriber
+     *
+     * @pslam-return ContainerInterface&\PHPUnit\Framework\MockObject\MockObject
+     */
+    private function buildContainer($subscriber) : ContainerInterface
+    {
+        $container = $this->createContainerMock();
+        $container->method('has')->with('config')->willReturn(true);
+        $container->method('get')->with('config')->willReturn(
+            [
+                'doctrine' => [
+                    'event_manager' => [
+                        'orm_default' => [
+                            'subscribers' => [$subscriber],
+                        ],
+                    ],
+                ],
+            ]
+        );
+
+        return $container;
+    }
+
+    /**
+     * @param mixed $listener
+     *
+     * @psalm-return ContainerInterface&\PHPUnit\Framework\MockObject\MockObject
+     */
+    private function buildContainerWithListener($listener) : ContainerInterface
+    {
+        $container = $this->createContainerMock();
+        $container->method('has')->with('config')->willReturn(true);
+        $container->method('get')->with('config')->willReturn($this->getConfigForListener($listener));
+
+        return $container;
+    }
+
+    /**
+     * @param mixed $subscriber
+     *
+     * @return array<string, mixed>
      */
-    private function buildContainer($subscriber) : ObjectProphecy
+    private function getConfigForSubscriber($subscriber) : array
     {
-        $container = $this->prophesize(ContainerInterface::class);
-        $container->has('config')->willReturn(true);
-        $container->get('config')->willReturn([
+        return [
             'doctrine' => [
                 'event_manager' => [
                     'orm_default' => [
                         'subscribers' => [$subscriber],
                     ],
                 ],
             ],
-        ]);
-
-        return $container;
+        ];
     }
 
     /**
      * @param mixed $listener
+     *
+     * @return array<string, mixed>
      */
-    private function buildContainerWithListener($listener) : ObjectProphecy
+    private function getConfigForListener($listener) : array
     {
-        $container = $this->prophesize(ContainerInterface::class);
-        $container->has('config')->willReturn(true);
-        $container->get('config')->willReturn([
+        return [
             'doctrine' => [
                 'event_manager' => [
                     'orm_default' => [
                         'listeners' => [$listener],
                     ],
                 ],
             ],
-        ]);
+        ];
+    }
 
-        return $container;
+    /**
+     * @psalm-return ContainerInterface&\PHPUnit\Framework\MockObject\MockObject
+     */
+    private function createContainerMock() : ContainerInterface
+    {
+        return $this->getMockBuilder(ContainerInterface::class)->onlyMethods(['has', 'get'])->getMock();
```

Comment:
`$this->createMock()` suffices

Reply:
Addressed with bbe09841c12999e74a99b7af5d99804871fc90f2